torch/jit/annotations.py

```diff
@@ -240,6 +242,21 @@ def as_ann(ann):
     return arg_types, return_type
 
 
+# Finds common type for enum values belonging to an Enum class. If not all
+# values have the same type, AnyType is returned.
+def get_enum_value_type(e: enum.Enum, loc):
+    enum_values = list(e)
+    if not enum_values:
+        raise ValueError("No enum values defined for: '{}'".format(e.__class__))
+
+    types = set([type(v.value) for v in enum_values])
+
+    # Likely path of all enum values having same type.
+    if len(types) == 1:
+        return try_ann_to_type(types.pop(), loc)
+    return AnyType.get()
```

Comment:
see comment above about supporting mutable types for enum value

Reply:
I think we should just throw an error instead. If users complain we can get back to this, but the common case is one type of value for an enum.

Reply:
Good point. Done.